internal/trace/parent/parent.go

```diff
@@ -0,0 +1,41 @@
+package parent
+
+import (
+	"context"
+
+	"go.opentelemetry.io/otel/api/core"
+	"go.opentelemetry.io/otel/api/key"
+	"go.opentelemetry.io/otel/api/trace"
+)
+
+func GetSpanContextAndLinks(ctx context.Context, ignoreContext bool) (core.SpanContext, bool, []trace.Link) {
+	lsctx := trace.SpanFromContext(ctx).SpanContext()
+	rsctx := trace.RemoteSpanContextFromContext(ctx)
+
+	if ignoreContext {
+		links := addLinkIfValid(nil, lsctx, "current")
+		links = addLinkIfValid(links, rsctx, "remote")
+
+		return core.EmptySpanContext(), false, links
+	}
+	if lsctx.IsValid() {
+		links := addLinkIfValid(nil, rsctx, "remote")
+		return lsctx, false, links
+	}
+	if rsctx.IsValid() {
+		return rsctx, true, nil
+	}
+	return core.EmptySpanContext(), false, nil
+}
+
+func addLinkIfValid(links []trace.Link, sc core.SpanContext, kind string) []trace.Link {
+	if !sc.IsValid() {
+		return links
+	}
+	return append(links, trace.Link{
+		SpanContext: sc,
+		Attributes: []core.KeyValue{
+			key.String("ignored-on-demand", kind),
```

Comment:
I am not sure about `ignored-on-demand`.  kind (key/value) should be defined in sdk-spec.
How about 'link-type'?

Reply:
Yeah, wasn't sure about the name, but I clearly wanted to preserve some information that we are not making some span context a parent of a span that's about to be created because of the `WithNewRoot()` option. So I wanted some information that tells that this remote/local span context could be a parent, but isn't, because the code creating the span decided so. Not sure if `link-type` conveys that. `possible-remote-parent` or `possible-local-parent`? There are quite a mouthful already.

Reply:
For now just open an issue.

Reply:
Good idea, will do.

Reply:
See #461.